zcash_primitives/src/transaction/mod.rs

```diff
@@ -955,9 +1166,18 @@ pub trait TransactionDigest<A: Authorization> {
 
     fn digest_orchard(
         &self,
-        orchard_bundle: Option<&orchard::Bundle<A::OrchardAuth, Amount>>,
+        orchard_bundle: Option<&orchard::Bundle<A::OrchardAuth, Amount, OrchardVanilla>>,
     ) -> Self::OrchardDigest;
 
+    #[cfg(zcash_unstable = "nu6" /* TODO nu7 */ )]
+    fn digest_orchard_zsa(
```

Comment:
Generalize `digest_orchard()` using generics

Reply:
It's hard because of Authorization, we use separate for Orchard and ZSA as we don't want to add type parameter to it